File: PeriodisationProgramApp.Domain/Entities/TrainingSession.cs
namespace PeriodisationProgramApp.Domain.Entities
{
    public class TrainingSession : Entity
    {
        public TrainingSession(int week, DayOfWeek dayOfWeek, int repsInReserve)
        {
            Week = week;
            DayOfWeek = dayOfWeek;
            RepsInReserve = repsInReserve;
        }

        public int Week { get; set; }

        public DayOfWeek DayOfWeek { get; set; }

        public int RepsInReserve { get; set; }

        public List<TrainingSessionExercise> Exercises { get; set; } = new();
    }
}

File: PeriodisationProgramApp.Web/src/types/enitities/TrainingSession.ts
import { DayOfWeek } from "../../enums/DayOfWeek";
import { BaseEntity } from "./BaseEntity";
import { TrainingSessionExercise } from "./TrainingSessionExercise";

export type TrainingSession = {
  week: number;
  dayOfWeek: DayOfWeek;
  repsInReserve: number;
  exercises: TrainingSessionExercise[];
} & BaseEntity;
